test/integration/factory.js

```diff
@@ -51,4 +56,56 @@ describe('factory', function() {
             return done();
         });
     });
+
+    it('invalid filters in createCacheTable', function(done) {
+        const TEST_SOURCE_TYPE = 'test-source';
+        const TestSource = Node.create(TEST_SOURCE_TYPE, {
+            table: Node.PARAM.STRING()
+        }, { cache: true });
+        TestSource.prototype.sql = function() {
+            return 'select * from ' + this.table;
+        };
+
+        const invalid_filters = {
+            price: {
+                type: 'range',
+                column: 'price',
+                params: {
+                }
+            }
+        };
+        const sourceDefinition = {
+            type: TEST_SOURCE_TYPE,
+            params: {
+                table: 'airbnb_rooms',
+            }
+        };
+        const samplingDefinition = {
+            type: 'sampling',
+            params: {
+                source: sourceDefinition,
+                sampling: 1,
+                seed: 12345,
+                filters: invalid_filters
+            }
+        };
+        const definition = {
+            type: 'sampling',
+            params: {
+                source: samplingDefinition,
```

Comment:
Do we need to nest two sampling nodes?

Reply:
Yes, I could only reproduce the exception within `createTableQuery` that way. With just one the exception occurs after createTableQuery (in getColumns) where it is already handled correctly.